File: examples/drivers/node/basic.ts
const BASE_URL = "http://localhost:8080";

interface HealthResponse {
  status: string;
  [key: string]: unknown;
}

interface MyRecord {
  id?: string;
  collection: string;
  data: Record<string, unknown>;
}

async function main(): Promise<void> {
  console.log("=== PrimusDB Node.js Example ===");

  try {
    // Check health
    const healthResp = await fetch(`${BASE_URL}/health`);
    const health: HealthResponse = await healthResp.json();
    console.log("Health:", JSON.stringify(health));
  } catch (e) {
    console.error("Error connecting to PrimusDB:", e);
    return;
  }

  try {
    // Get version
    const versionResp = await fetch(`${BASE_URL}/version`);
    const version = await versionResp.json();
    console.log("Version:", JSON.stringify(version));
  } catch (e) {
    console.error("Error fetching version:", e);
  }

  try {
    // Create a record
    const record: MyRecord = {
      collection: "users",
      data: {
        name: "Alice",
        email: "alice@example.com",
      },
    };
    const createResp = await fetch(`${BASE_URL}/records`, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(record),
    });
    const created = await createResp.json();
    console.log("Created record:", JSON.stringify(created));
  } catch (e) {
    console.error("Error creating record:", e);
  }

  try {
    // Query records
    const queryResp = await fetch(`${BASE_URL}/records/users`);
    const records = await queryResp.json();
    console.log("Records:", JSON.stringify(records));
  } catch (e) {
    console.error("Error querying records:", e);
  }
}

main();

File: examples/drivers/ruby/basic.rb
require "net/http"
require "json"
require "uri"

BASE_URL = "http://localhost:8080"

puts "=== PrimusDB Ruby Example ==="

# Check health
begin
  uri = URI("#{BASE_URL}/health")
  resp = Net::HTTP.get_response(uri)
  puts "Health: #{resp.body}"
rescue StandardError => e
  puts "Error connecting to PrimusDB: #{e.message}"
  exit 1
end

# Get version
begin
  uri = URI("#{BASE_URL}/version")
  resp = Net::HTTP.get_response(uri)
  puts "Version: #{resp.body}"
rescue StandardError => e
  puts "Error fetching version: #{e.message}"
end

# Create a record
begin
  uri = URI("#{BASE_URL}/records")
  record = {
    collection: "users",
    data: {
      name: "Alice",
      email: "alice@example.com"
    }
  }
  http = Net::HTTP.new(uri.host, uri.port)
  req = Net::HTTP::Post.new(uri.path)
  req["Content-Type"] = "application/json"
  req.body = record.to_json
  resp = http.request(req)
  puts "Created record: #{resp.body}"
rescue StandardError => e
  puts "Error creating record: #{e.message}"
end

# Query records
begin
  uri = URI("#{BASE_URL}/records/users")
  resp = Net::HTTP.get_response(uri)
  puts "Records: #{resp.body}"
rescue StandardError => e
  puts "Error querying records: #{e.message}"
end
